src/lib/ndpi_main.c

```diff
@@ -1751,7 +1749,6 @@ static int ndpi_add_host_ip_subprotocol(struct ndpi_detection_module_struct *ndp
   struct in_addr pin;
   
   inet_pton(AF_INET, value, &pin);
-  pin.s_addr = ntohl(pin.s_addr);
```

Comment:
`inet_pton` already puts `pin` into network byte order, that is the default behavior. No need to manually convert it again.